classes/UpgradeTools/ModuleAdapter.php

```diff
@@ -80,17 +108,32 @@ public function getModuleDataUpdater()
         return $this->moduleDataUpdater;
     }
 
+    /**
+     * @return ModuleRepository
+     */
+    public function getModuleRepository()
+    {
+        return $this->moduleRepository;
+    }
+
+    /**
+     * @return ModuleDisabler
+     */
+    public function getModuleDisabler()
+    {
+        return $this->moduleDisabler;
+    }
+
     /**
      * Upgrade action, disabling all modules not made by PrestaShop.
-     *
-     * It seems the 1.6 version of is the safest, as it does not actually load the modules.
-     *
-     * @param string $pathToUpgradeScripts Path to the PHP Upgrade scripts
      */
-    public function disableNonNativeModules($pathToUpgradeScripts)
+    public function disableNonNativeModules()
     {
-        require_once $pathToUpgradeScripts . 'deactivate_custom_modules.php';
-        deactivate_custom_modules();
+        $customModules = $this->moduleRepository->getCustomModulesOnDisk([$this->originVersion, $this->upgradeVersion]);
+        foreach ($customModules as $moduleName) {
+            $this->moduleDisabler->disableModuleFromDatabase($moduleName);
```

Comment:
Since this class is not final, always prefer getter function
```suggestion
            $this->getModuleDisabler()->disableModuleFromDatabase($moduleName);
```

Reply:
In case it's overridden?

Reply:
Yes, even for us we we have another Disabler based on this class. If we have setter / getter, always prefer them.